pkg/config/config.go

```diff
@@ -146,3 +146,36 @@ func GetOS() []string {
 	}
 	return res
 }
+
+// getConfigPath returns the path to the configuration directory respecting
+// the `XDG Base Directory specification` using the following strategy:
+//   - to prevent breaking of existing configurations, check if "$HOME/.bin/config.json"
+//     exists and return "$HOME/.bin"
+//   - if "XDG_CONFIG_HOME" is set, return "$XDG_CONFIG_HOME/bin"
+//   - if "$HOME/.config" exists, return "$home/.config/bin"
+//   - default to "$HOME/.bin/"
+// ToDo: move the function to config_unix.go and add a similar function for windows,
+//       %APPDATA% might be the right place on windows
+func getConfigPath() (string, error) {
+	home, homeErr := os.UserHomeDir()
+	if homeErr == nil {
+		if _, err := os.Stat(path.Join(home, ".bin", "config.json")); !os.IsNotExist(err) {
```

Comment:
Shouldn't this be `filepath.Join` as well (instead of `path.Join`)?
Also on the next line.